nixos/modules/services/databases/postgresql.nix

```diff
@@ -306,8 +346,9 @@ in
               ln -sfn "${pkgs.writeText "recovery.conf" cfg.recoveryConfig}" \
                 "${cfg.dataDir}/recovery.conf"
             ''}
+            chmod ${dirMode} "${cfg.dataDir}"
```

Comment:
Is it necessary to re-apply the mode at each startup?
If so, this should perhaps be applied recursively.
This might be a costly operation on big databases though.

Reply:
PostgreSQL is sensitive only for group mode of dataDir itself. Files inside can be even world readable, but top-level dataDir may be only 0700 or 0750 (for pg11). Otherwise it just doesn't start.

Reply:
@pacien ok, I made it optional. Now `groupAccess` is tristate with `null` being "don't change mask" (previous behavior).

Reply:
How does PostgreSQL apply the right permissions to new files within the data directory?
Is it copying the mode of dataDir itself?